best-practices/native/react-native/common-components/phone-number.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import { View, Text, StyleSheet, Linking } from 'react-native';
+import PropTypes from 'prop-types';
+import { connectActionSheet } from '@expo/react-native-action-sheet';
+
+// Delete this comment when using this code in project
+// ---------------------------------------------------
+// This component is passed a phone number string, along with a few styling properties.
+// It returns text that when pressed, opens an action sheet giving the user an option to either call or text that number
+// This requires use of the package "@expo/react-native-action-sheet".
+
+const styles = StyleSheet.create({
+  phoneText: {
+      // If you have a desired font loaded, place it here as the "fontFamily" property.
+  }
+});
+
+class PhoneNumber extends Component {
+
+  _onOpenActionSheet = (phoneNo) => {
+    // Same interface as https://facebook.github.io/react-native/docs/actionsheetios.html
+    const options = [`Call:  ${phoneNo}`, `Text:  ${phoneNo}`, 'Cancel'];
+    const cancelButtonIndex = 2;
+
+    this.props.showActionSheetWithOptions(
+      {
+        options,
+        cancelButtonIndex,
+      },
+      buttonIndex => {
+        // Do something here depending on the button index selected
+        if (buttonIndex === 0) Linking.openURL(`tel:${phoneNo}`);
+        if (buttonIndex === 1) Linking.openURL(`sms:${phoneNo}`);
+      },
+    );
+  };
+
+  render() {
+    const { phoneNo, disallowFontScaling } = this.props;
+    const { fontSize, color, lineHeight } = this.props;
+
+    return (
+      <View>
+        <Text allowFontScaling={!disallowFontScaling} style={[styles.phoneText, { fontSize, color, lineHeight }]} onPress={() => this._onOpenActionSheet(phoneNo)}>{phoneNo}</Text>
+      </View>
+    );
+    
+  }
+}
```

Comment:
I really wish we could set up some unit tests for this since this is going to be in many applications.

Reply:
I second this. Can we get a simple [Jest](https://jestjs.io/docs/en/tutorial-react) for this component? even just I/O testing would be great.